### ISSUE TITLE
PMD "broke" when use 'as user'

### ISSUE THREAD

**Issue by pablogomez2197** [pmd/pmd#4944]
[apex]
**Affects PMD Version:** 
v7.0.0.rc4
**Rule:**
All
Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_XXX_XXX.html#XXX>

**Description:**
I have an apex class where I have a lot apex warning/problems, but when i put 'as user' in the code all of them dissapears.
**Code Sample demonstrating the issue:**
<img width="733" alt="image" src="https://github.com/pmd/pmd/assets/110610165/2bb3280b-c386-43c8-a070-1b5c69aa1936">
and then:
<img width="733" alt="image" src="https://github.com/pmd/pmd/assets/110610165/06756835-b131-47b5-8486-0b7b1849d2aa">

```java

```

**Expected outcome:**
The expeted outcome it's the warning and problems, all of them.
PMD should report a violation at line ..., but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by oowekyala** [comment]
This is fixed in PMD 7.0.0, see #3973

**Comment by pablogomez2197** [comment]
We have that version installed and it keeps happening to us.
